### ISSUE TITLE
Custom setting of v1beta1 batcher seems to be ineffective

### ISSUE THREAD

**Issue by Peilun-Li** [kubeflow/kfserving#1666]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
(While things work fine in v0.4/v1alpha2) Customization of the v1beta1 batcher setting (maxBatchSize, maxLatency, timeout) seems to be ineffective -- the actual honored value is still the default value (maxBatchSize=32, maxLatency=5000)

Here I just use the [v1beta1 batcher example](https://github.com/kubeflow/kfserving/blob/master/docs/samples/batcher/basic/batcher_v1beta1.yaml) provided in this repo, but this problem remains if I use other custom images.

Steps:
1. `kubectl apply -f batcher_v1beta1.yaml` (The desired max latency should then be around ~300ms)
2. curl the endpoint 
```
$ curl -v https://pytorch-cifar10.namespace.example.com/v1/models/pytorch-cifar10:predict -d @./input.json

< x-envoy-upstream-service-time: 5008
<
* Connection #0 to host pytorch-cifar10.namespace.example.com left intact
{"message":"","batchId":"d8a42eae-05bb-4b41-bacd-60916544c8f1","predictions":[[-1.6099601984024048,-2.6461071968078613,0.3284444510936737,2.4825074672698975,0.4352457523345947,2.3108041286468506,1.0005676746368408,-0.42327630519866943,-0.5100943446159363,-1.7978390455245972]]}* Closing connection 0
```
As can be seen it still takes the default 5000ms latency within the batcher

More info

main container works fine
``` 
$ kubectl logs pytorch-cifar10-predictor-default-9vl99-deployment-59dddb7m7f7x kfserving-container
[I 210616 22:22:13 storage:45] Copying contents of /mnt/models to local
[I 210616 22:22:13 storage:223] Linking: /mnt/models/cifar10.py to pytorch-cifar10/cifar10.py
[I 210616 22:22:13 storage:223] Linking: /mnt/models/model.pt to pytorch-cifar10/model.pt
[I 210616 22:22:13 kfserver:115] Registering model: pytorch-cifar10
[I 210616 22:22:13 kfserver:96] Listening on port 8080
[I 210616 22:22:13 kfserver:98] Will fork 1 workers
[I 210616 22:36:37 web:2243] 200 POST /v1/models/pytorch-cifar10:predict (127.0.0.1) 2.37ms
```
agent still has the default 5000ms latency
``` 
$ kubectl logs pytorch-cifar10-predictor-default-9vl99-deployment-59dddb7m7f7x agent
{"level":"info","ts":"2021-06-16T22:22:13.356Z","caller":"logging/config.go:110","msg":"Successfully created the logger."}
{"level":"info","ts":"2021-06-16T22:22:13.356Z","caller":"logging/config.go:111","msg":"Logging level set to: info"}
{"level":"info","ts":"2021-06-16T22:22:13.356Z","caller":"logging/config.go:78","msg":"Fetch GitHub commit ID from kodata failed","error":"\"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":"2021-06-16T22:22:13.356Z","caller":"batcher/handler.go:157","msg":"Starting batch loop"}
{"level":"info","ts":"2021-06-16T22:36:32.716Z","caller":"batcher/handler.go:242","msg":"serving request /v1/models/pytorch-cifar10:predict"}
```

pod description has the expected args passed in 
``` 
$ kubectl describe pod pytorch-cifar10-predictor-default-9vl99-deployment-59dddb7m7f7x

  agent:
    Container ID:  docker://d19a1d10c8962a23b9bc1bd66e8b3b155fd8950705b50ca6db2ce4d2abfc022b
    Image:         kfserving/agent:v0.5.1
    Image ID:      docker-pullable://kfserving/agent@sha256:8656bf9be31c491bc6e6ae2c4943b5bfb5bc58e9f807c8695dbbc21477c9e78e
    Port:          <none>
    Host Port:     <none>
    Args:
      --enable-batcher
      true
      --max-batchsize
      16
      --max-latency
      300
    State:          Running
      Started:      Wed, 16 Jun 2021 15:22:13 -0700
    Ready:          True
```

isvc  description has the expected settings passed in 
```
$ kubectl describe isvc pytorch-cifar10
Name:         pytorch-cifar10

Labels:       <none>
Annotations:  <none>
API Version:  serving.kubeflow.org/v1beta1
Kind:         InferenceService
Metadata:
  Creation Timestamp:  2021-06-16T22:21:32Z
  Finalizers:
    inferenceservice.finalizers
  Generation:        1
  Resource Version:  307040044
  UID:               10be7acf-397d-4e94-865b-f2732a419f14
Spec:
  Predictor:
    Batcher:
      Max Batch Size:  16
      Max Latency:     300
    Min Replicas:      1
    Pytorch:
      Model Class Name:  Net
      Name:              kfserving-container
      Protocol Version:  v1
      Resources:
        Limits:
          Cpu:     1
          Memory:  2Gi
        Requests:
          Cpu:          1
          Memory:       2Gi
      Runtime Version:  v0.5.1
      Storage Uri:      gs://kfserving-samples/models/pytorch/cifar10/
```

The same issue also applies to max batch size, e.g., setting a low value of 2 won't prevent the batcher to send a request to kfserving-container with 32 batch size.

**What did you expect to happen:**
The v1beta1 batcher should honor the customization of latency, batch size, etc.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KFServing Version: 0.5.1
- Kubeflow version:
- Kfdef:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

**Comment by yuzisun** [comment]
@Peilun-Li This is fixed already, should be available in 0.6.0.

**Comment by yuzisun** [comment]
released in 0.6.0
